File: admin-web/src/types/appointment.ts
import type { PageQuery } from './common';

export interface AppointmentVO {
  id: number;
  estateId?: number;
  buildingId?: number;
  buildingName?: string;
  unitId?: number;
  unitName?: string;
  roomId?: number;
  roomNo?: string;
  userId?: number;
  customerUserId?: number;
  customerName?: string;
  salesUserId?: number;
  salesName?: string;
  appointmentTime?: string;
  contactName?: string;
  contactPhone?: string;
  remark?: string;
  cancelReason?: string;
  status: number;
  createTime?: string;
}

export interface AppointmentPageQuery extends PageQuery {
  estateId?: number | string;
  buildingId?: number | string;
  buildingName?: string;
  salesUserId?: number | string;
  salesName?: string;
  contactName?: string;
  contactPhone?: string;
  status?: number | '';
}

export interface AppointmentStatusUpdate {
  id: number;
  status: number;
  salesUserId?: number;
}

export interface SalesUserVO {
  id: number;
  username: string;
  realName?: string;
  phone?: string;
}

File: backend/smart-sales-control/src/main/java/com/sellinghouses/salescontrol/module/appointment/entity/Appointment.java
package com.sellinghouses.salescontrol.module.appointment.entity;

import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Appointment {

    private Long id;

    private Long userId;

    private Long salesUserId;

    private Long roomId;

    private LocalDateTime appointmentTime;

    private String contactName;

    private String contactPhone;

    private String remark;

    private Integer status;

    private String cancelReason;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    private Long createUser;

    private Long updateUser;
}
